File: GoWebApi/app/models/user_type.go
package models

// UserType 用户类型
// 使用定义类型而非别名, 避免与普通 int 混用导致赋值越界
type UserType int

const (
	UserTypeOrdinary UserType = 1 // 普通用户
)

// IsUserTypeValid 判断用户类型是否有效
func IsUserTypeValid(t UserType) bool {
	switch t {
	case UserTypeOrdinary:
		return true
	}
	return false
}

// GetUserTypeText 获取用户类型中文描述
func GetUserTypeText(t UserType) string {
	switch t {
	case UserTypeOrdinary:
		return "普通用户"
	default:
		return ""
	}
}

File: RsWebApi/src/app/models/user_type.rs
//! 用户类型 (对应 Go 项目 app/models/user_type.go)
//!
//! 以下函数当前业务未使用, 保留以与 Go 版代码结构对齐 (Go 端同样提供这些辅助函数);
//! 由 `#![allow(dead_code)]` 抑制编译告警。
#![allow(dead_code)]

/// 普通用户
pub const USER_TYPE_ORDINARY: i32 = 1;

/// 判断用户类型是否有效
pub fn is_user_type_valid(t: i32) -> bool {
    matches!(t, USER_TYPE_ORDINARY)
}

/// 获取用户类型中文描述
pub fn get_user_type_text(t: i32) -> &'static str {
    match t {
        USER_TYPE_ORDINARY => "普通用户",
        _ => "",
    }
}
